### PR TITLE
messagebus spec

### PR DESCRIPTION
**Pull request by pelikhan**
Allows CODAL-like messages over jacdac.

### CODE REVIEW

File: services/messagebus.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Message Bus
+
+    identifier: 0x17feef9b
+    
+A lightweight event bus that allows to broadcast
+short messages.
+
+## Commands
+
+    command send @ 1 {
```
Comment (mmoskal) [inline services/messagebus.md:10]: this probably needs to be @ 0x80 - you should be getting an error here?

File: services/messagebus.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        id: u16;
```
Comment (mmoskal) [inline services/messagebus.md:11]: I'm not sure we accept semicolons, we surely don't require them

Comment (mmoskal) [reply services/messagebus.md:11]: also, probably best to use `u32` for future extensibility

File: services/messagebus.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        name: u16;
+    }
+
+## Events
+
+    event message @ 2 {
```
Comment (mmoskal) [inline services/messagebus.md:17]: events have a separate numbers from commands, so you can start at `1`

File: services/messagebus.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Message Bus
+
+    identifier: 0x17feef9b
+    
+A lightweight event bus that allows to broadcast
+short messages.
```
Comment (mmoskal) [inline services/messagebus.md:6]: Would be nice to have some usage scenario. I'm actually not sure about it.

File: services/messagebus.md (hunk 1 above)
Comment (pelikhan) [reply services/messagebus.md:10]: will check

File: services/messagebus.md [hunk 5]
```diff
@@ -3,20 +3,18 @@
     identifier: 0x17feef9b
     
 A lightweight event bus that allows to broadcast
-short messages.
+short messages, similar to the Scratch message APIs.
```
Comment (mmoskal) [inline services/messagebus.md:6]: Is the service just going to raise the event after it gets a send?

Comment (pelikhan) [reply services/messagebus.md:6]: so basically the idea is that when the device receives a JACDAC event; it raises it on the CODAL event bus side.

Comment (mmoskal) [reply services/messagebus.md:6]: so there is no generally-agreed-upon meaning to these events? If it's for CODAL, it should be called CODAL Message Bus; but I think we also need a clear usage scenario.

**Pull request closed without merging**